SConstruct

```diff
@@ -586,25 +611,127 @@ def list_sources(base, suffixes):
+    result.append([filenames, kwds])
+
+def get_soci_sources(style):
+    result = []
     cpp_path = [
         'src/soci/src/core',
         'src/sqlite',]
-    object_builder.add_source_files(
+    append_sources(result,
         'src/ripple/unity/soci.cpp',
         CPPPATH = cpp_path)
-    if is_unity:
-        object_builder.add_source_files(
+    if style == 'unity':
+        append_sources(result,
             'src/ripple/unity/soci_ripple.cpp',
             CPPPATH = cpp_path)
+    return result
+
+def get_classic_sources():
+    result = []
+    append_sources(result,
+        *list_sources('src/ripple/app', '.cpp'),
+        CPPPATH=[
+             'src/soci/src/core',
+             'src/sqlite',]
+    )
+    append_sources(result, *list_sources('src/ripple/basics', '.cpp'))
+    append_sources(result, *list_sources('src/ripple/core', '.cpp'))
+    append_sources(result, *list_sources('src/ripple/crypto', '.cpp'))
+    append_sources(result, *list_sources('src/ripple/json', '.cpp'))
+    append_sources(result, *list_sources('src/ripple/legacy', '.cpp'))
+    append_sources(result, *list_sources('src/ripple/net', '.cpp'))
+    append_sources(result, *list_sources('src/ripple/overlay', '.cpp'))
+    append_sources(result, *list_sources('src/ripple/peerfinder', '.cpp'))
+    append_sources(result, *list_sources('src/ripple/protocol', '.cpp'))
+    append_sources(result, *list_sources('src/ripple/shamap', '.cpp'))
+    append_sources(result, *list_sources('src/ripple/nodestore', '.cpp'),
+        CPPPATH=[
+            'src/rocksdb2/include',
+            'src/snappy/snappy',
+            'src/snappy/config',
+        ])
+
+    result += get_soci_sources('classic')
+    return result
+
+def get_unity_sources():
+    result = []
+    append_sources(
+        result,
+        'src/ripple/unity/app.cpp',
+        'src/ripple/unity/app1.cpp',
+        'src/ripple/unity/app2.cpp',
+        'src/ripple/unity/app3.cpp',
+        'src/ripple/unity/app4.cpp',
+        'src/ripple/unity/app5.cpp',
+        'src/ripple/unity/app6.cpp',
+        'src/ripple/unity/app7.cpp',
+        'src/ripple/unity/app8.cpp',
+        'src/ripple/unity/app9.cpp',
+        'src/ripple/unity/core.cpp',
+        'src/ripple/unity/basics.cpp',
+        'src/ripple/unity/crypto.cpp',
+        'src/ripple/unity/net.cpp',
+        'src/ripple/unity/overlay.cpp',
+        'src/ripple/unity/peerfinder.cpp',
+        'src/ripple/unity/json.cpp',
+        'src/ripple/unity/protocol.cpp',
+        'src/ripple/unity/shamap.cpp',
+        'src/ripple/unity/legacy.cpp',
+    )
+
+    result += get_soci_sources('unity')
+
+    append_sources(
+        result,
+        'src/ripple/unity/nodestore.cpp',
+        CPPPATH=[
+            'src/rocksdb2/include',
+            'src/snappy/snappy',
+            'src/snappy/config',
+        ])
+
+    return result
 
 # Declare the targets
 aliases = collections.defaultdict(list)
 msvc_configs = []
 
+def should_prepare_target (cl_target,
```

Comment:
This indentation is also legal!  However, this file mostly consistently uses the style on line 688, e.g.